mixed-content/generic/expect.py

```diff
@@ -0,0 +1,99 @@
+import json, os, urllib, urlparse
+
+def redirect(url, response):
+    response.add_required_headers = False
+    response.writer.write_status(301)
+    response.writer.write_header("access-control-allow-origin", "*")
+    response.writer.write_header("location", url)
+    response.writer.end_headers()
+    response.writer.write("")
+
+def create_redirect_url(request, swap_scheme = False):
+    parsed = urlparse.urlsplit(request.url)
+    destination_netloc = parsed.netloc
+    scheme = parsed.scheme
+
+    if swap_scheme:
+      if parsed.scheme == "http":
```

Comment:
What about HTTPS -> HTTP?

This could be done as the following, for example:

```
if swap_scheme:
  scheme = "http" if parsed.scheme == "https" else "https"
```